python/tidy/servo_tidy/tidy.py

```diff
@@ -723,13 +724,40 @@ def parse_config(content):
     config['ignore']['directories'] = map(lambda path: os.path.join(*path.split('/')),
                                           config['ignore']['directories'])
 
+    # Add dict of dir, list of expected ext to config
+    dirs_to_check = config_file.get("check_ext", {})
+    # Fix the paths (OS-dependent)
+    for path, exts in dirs_to_check.items():
+        fixed_path = os.path.join(*path.split('/'))
+        config['check_ext'][fixed_path] = exts
+
     # Override default configs
     user_configs = config_file.get("configs", [])
     for pref in user_configs:
         if pref in config:
             config[pref] = user_configs[pref]
 
 
+def check_directory_files(directories, print_text=True):
+    if print_text:
+        print '\rChecking directories for correct file extensions...'
+    for directory, file_extensions in directories.items():
+        files = os.listdir(directory)
+        for filename in files:
+            full_path = os.path.join(directory, filename)
+            if not os.path.isfile(full_path):
+                continue
```

Comment:
yes, that might be a good idea. Actually I was wondering if you meant: "making sure we're only checking extensions for actual files", or rather "besides checking file extensions, let's also check if those are files in the first place". 

If it's not a file (I checked the docs and symlinks are considered files for this function, to be honest I do not know what would be such a non-file 😄 ), should we really throw an error? Or do you mean yielding an error like is done with incorrect extensions? If so, what should be the error message? Just the same as for wrong extensions?